### ISSUE TITLE
Simplify state management

### ISSUE THREAD

**Issue by hannobraun** [braun-robotics/rust-lpc82x-hal#69]
Currently, all peripherals have an initial `Unknown` state, to model the fact that the HAL API doesn't know what happened to the hardware before the API was initialized. Other types, like pins, fixed functions, and movable functions, have something similar.

There are issues with that:
- It's needlessly complicated. It would be nicer if the user wouldn't have to deal with that whole concept.
- In the case of peripherals, it seems especially unnecessary. With one additional instruction, we could just reset the peripheral and create a known state (unless I'm misunderstanding how peripheral reset works, which is possible).
- In the case of the other types, the user needs to call an unsafe method to promise the API that the hardware is still in its initial state. This seems complicated and error-prone. Plus, requiring `unsafe` to use the API is not good.

I initially designed it that way because I wanted to avoid as much overhead as possible, but as I gain more experience with the API, this seems more and more ridiculous. The overhead of creating a known state should be negligible. If I'm wrong about that, I'm sure better (and optional) solutions can be found on a case-by-case basis.

**Comment by hannobraun** [comment]
Another problem with the current situation that I forgot to mention: Implementation complexity. Replacing that state management code with a few method calls that create a known initial state should result in simpler code.

**Comment by hannobraun** [comment]
I've made changes to peripheral states in #79. I've decided to not remove the `Unknown` state as originally planned. I didn't like the API changes that it required (need to pass `syscon::Handle` into every constructor), and I wasn't sure if the peripheral reset code I wrote worked as expected.

I'm no longer sure it would be worth it, but if we attempt this again, we need some better testing infrastructure in place to make sure it works correctly.

In any case, I haven't touched states for pins, fixed functions and movable functions so far. Removing the need for `unsafe` there is still a legitimate concern.
